Dockerfile

```diff
@@ -19,17 +19,41 @@
 ## 1. Base (stable)                                                          ##
 ###############################################################################
 
-FROM python:2.7
+FROM python:2.7-slim
 MAINTAINER CERN <info@invenio-software.org>
 
 # nodejs repo
 # detects distribution and adds the right repo
 # See: https://github.com/joyent/node/wiki/Installing-Node.js-via-package-manager#debian-and-ubuntu-based-linux-distributions
-RUN curl -sL https://deb.nodesource.com/setup | bash -
+RUN apt-get update && \
+    apt-get -qy install --fix-missing --no-install-recommends \
+        curl \
+        && \
+    curl -sL https://deb.nodesource.com/setup | bash -
 
 # install requirements from repos
+# also clean up apt
 RUN apt-get update && \
-    apt-get -qy install poppler-utils git subversion nodejs mysql-client sudo --fix-missing
+    apt-get -qy upgrade --fix-missing --no-install-recommends && \
+    apt-get -qy install --fix-missing --no-install-recommends \
+        gcc \
+        git \
+        libffi-dev \
+        libmysqlclient-dev \
+        libssl-dev \
+        libxslt-dev \
+        mysql-client \
+        nodejs \
+        poppler-utils \
+        subversion \
+        sudo \
+        && \
+    apt-get clean autoclean && \
+    apt-get autoremove -y && \
+    rm -rf /var/lib/{apt,dpkg}/ && \
+    (find /usr/share/doc -depth -type f ! -name copyright|xargs rm || true) && \
```

Comment:
I would consider `-delete` instead of passing to `xargs`, which can run into troubles if the script hits the arg count limit.